frontend/src/app/main/ui/share_link.cljs

```diff
@@ -0,0 +1,232 @@
+
+        copy-link
+        (fn [_]
+          (wapi/write-to-clipboard @link)
+          (st/emit! (dm/show {:type :info
+                              :content (tr "common.share-link.link-copied-success")
+                              :timeout 3000})))
+
+        try-delete-link
+        (fn [_]
+          (reset! confirm true))
+
+        delete-link
+        (fn [_]
+          (let [params (prepare-params @opts)
+                slink  (d/seek #(= (:flags %) (:flags params)) slinks)]
+            (reset! confirm false)
+            (st/emit! (dc/delete-share-link slink)
+                      (dm/show {:type :info
+                                :content (tr "common.share-link.link-deleted-success")
+                                :timeout 3000}))))
+        ]
+
+    (mf/use-effect
+     (mf/deps file slinks @opts)
+     (fn []
+       (let [{:keys [flags pages] :as params} (prepare-params @opts)
+             slink  (d/seek #(and (= (:flags %) flags) (= (:pages %) pages)) slinks)
+             href   (when slink
+                      (let [pparams (:path-params route)
+                            qparams (-> (:query-params route)
+                                        (assoc :share-id (:id slink))
+                                        (assoc :index "0"))
+                            href    (rt/resolve router :viewer pparams qparams)]
+                        (assoc cf/public-uri :fragment href)))]
+         (reset! link (some-> href str)))))
+
+    [:div.modal-overlay
+     [:div.modal-container.share-link-dialog
+      [:div.modal-content
+       [:div.title
+        [:h2 (tr "common.share-link.title")]
+        [:div.modal-close-button
+         {:on-click close
+          :title (tr "labels.close")}
+         i/close]]
+
+       [:div.share-link-section
+        [:label (tr "labels.link")]
+        [:div.custom-input.with-icon
+         [:input {:type "text" :value (or @link "") :read-only true}]
+         [:div.help-icon {:title (tr "labels.copy")
+                          :on-click copy-link}
+          i/copy]]
+
+        [:div.hint (tr "common.share-link.permissions-hint")]]]
+
+      [:div.modal-content
+       (let [sections (:sections @opts)]
+         [:div.access-mode
+          [:div.title (tr "common.share-link.permissions-can-access")]
+          [:div.items
+           [:div.input-checkbox.check-primary.disabled
+            [:input.check-primary.input-checkbox {:type "checkbox" :disabled true}]
+            [:label (tr "labels.workspace")]]
+
+           [:div.input-checkbox.check-primary
+            [:input {:type "checkbox"
+                     :default-checked (contains? sections "viewer")}]
+            [:label (tr "labels.viewer")
+             [:span.hint "(" (tr "labels.default") ")"]]]
+
+           ;; [:div.input-checkbox.check-primary
+           ;;  [:input.check-primary.input-checkbox {:type "checkbox"}]
+           ;;  [:label "Handsoff" ]]
+           ]])
```

Comment:
I understand this is a work in progress, isnt't it?

Reply:
yes, in a future we will have more options there.